challenges/Back-End Challenge/api/routes/data.js

```diff
@@ -0,0 +1,147 @@
+const express = require('express');
+
+// a way to package routes together under one name
+const router = express.Router();
+
+// variable so that we can access the same list that is submitted
+let numbers;
+
+// this is to sort the list numerically instead of alphabetically
+function numberSort(a, b) {
+    return a - b;
+}
+
+//even though it should be: 
+        //router.get('/data')
+// this will be packaged as data and the base URL for every URL inside this file
+// will be /data. Therefore any route inside will be appending to /data
+
+
+
+
+// get request to resturn the current list
+router.get('/', (req, res, next) => {
+
+    // checks if the list has been created
+    if (typeof numbers !== 'undefined') {
+        res.status(200).json({
+            list: numbers
+        });
+    } else {
+        res.status(400).json({
+            message: 'No current list to return.'
+        });
+    }
+
+});
+
+
+// post request to create new list
+router.post('/', (req, res, next) => {
+
+     // checks if only one field in JSON object
+    if (Object.keys(req.body).length != 1){
```

Comment:
Oh yeah! I always plan to use !== or === for that very reason, maybe I didn't press down on the key all the way. Since the code didn't throw any errors I just assumed it was !==.

Reply:
Since I have my own particular standards for code, I like including a linter in personal projects that checks basic stuff like this so I don't violate my own rules. It's a small add for a lot of value.

Reply:
Nice. I will look into that!